packages/schemas/src/text/uiRender.ts

```diff
@@ -119,45 +120,56 @@ export const uiRender = async (arg: UIRenderProps<TextSchema>) => {
     textAlign: schema.alignment ?? DEFAULT_ALIGNMENT,
     whiteSpace: 'pre-wrap',
     wordBreak: 'break-word',
+    // Block layout styles
+    resize: 'none',
+    border: 'none',
+    outline: 'none',
+    marginBottom: bottomAdjustment + 'px',
+    paddingTop: topAdjustment + 'px',
+    backgroundColor: 'transparent',
   };
+  const textBlock = document.createElement('div');
+  Object.assign(textBlock.style, textBlockStyle);
 
   if (mode === 'form' || mode === 'designer') {
-    const textarea = document.createElement('textarea');
-    const textareaStyle: CSS.Properties = {
-      padding: 0,
-      resize: 'none',
-      border: 'none',
-      outline: 'none',
-      paddingTop: topAdjustment + 'px',
-      backgroundColor: 'transparent',
-      width: '100%',
-      height: '100%',
-    };
-    Object.assign(textarea.style, textareaStyle, fontStyles);
-    textarea.rows = 1;
-    textarea.placeholder = placeholder || '';
-    textarea.tabIndex = tabIndex || 0;
-
-    textarea.addEventListener(
-      'change',
-      (e: Event) => onChange && onChange((e.target as HTMLTextAreaElement).value)
+    textBlock.contentEditable = 'plaintext-only'
+    textBlock.tabIndex = tabIndex || 0
+    textBlock.innerText = value;
+    textBlock.addEventListener(
+      'blur',
+      (e: Event) => {
+        onChange && onChange((e.target as HTMLDivElement).innerText);
+        stopEditing && stopEditing();
+      }
     );
-    textarea.addEventListener('blur', () => stopEditing && stopEditing());
-    textarea.value = value;
-    container.appendChild(textarea);
+    if (placeholder) {
+      textBlock.setAttribute('dataPlaceholder', placeholder);
+      const placeholderStyle = document.createElement('style');
+      placeholderStyle.textContent = `
+        [contenteditable=true]:empty:before {
+          content: attr(placeholder);
+          pointer-events: none;
+          display: block;
+        }
+      `;
+      container.appendChild(placeholderStyle);
+    }
```

Comment:
There was an implementation for `placeholder` before, so I have added this to maintain the behaviour, although I'm not sure if it's possible to use it?

Reply:
I checked, but the placeholder disappears. We need to think of a way to display the placeholder in the div.